specs/Standards/MultiToken/README.md

```diff
@@ -0,0 +1,539 @@
+    ///    2^53, and therefore representable as JSON. See Approval Management
+    ///    standard for full explanation. Must have same length as token_ids
+    /// * `memo` (optional): for use cases that may benefit from indexing or
+    ///    providing information for a transfer
+
+    fn mt_batch_transfer(
+        &mut self,
+        receiver_id: AccountId,
+        token_ids: Vec<TokenId>,
+        amounts: Vec<U128>,
+        memo: Option<String>,
+    );
+    /// Batch transfer token/s and call a method on a receiver contract. A successful
+    /// workflow will end in a success execution outcome to the callback on the MultiToken
+    /// contract at the method `mt_resolve_batch_transfer`.
+    ///
+    /// You can think of this as being similar to attaching  tokens to a
+    /// function call. It allows you to attach any Fungible or Non Fungible Token in a call to a
+    /// receiver contract.
+    ///
+    /// Requirements:
+    /// * Caller of the method must attach a deposit of 1 yoctoⓃ for security
+    ///   purposes
+    /// * Contract MUST panic if called by someone other than token owner or,
+    ///   if using Approval Management, one of the approved accounts
+    /// * The receiving contract must implement `mt_on_transfer` according to the
+    ///   standard. If it does not, MultiToken contract's `mt_resolve_batch_transfer` MUST deal
+    ///   with the resulting failed cross-contract call and roll back the transfer.
+    /// * Contract MUST implement the behavior described in `mt_resolve_batch_transfer`
+    /// * `approval_id` is for use with Approval Management extension, see
+    ///   that document for full explanation.
+    /// * If using Approval Management, contract MUST nullify approved accounts on
+    ///   successful transfer.
+    ///
+    /// Arguments:
+    /// * `receiver_id`: the valid NEAR account receiving the token.
+    /// * `token_ids`: the tokens to transfer
+    /// * `amounts`: the amount of tokens to transfer for corresponding token_id
+    /// * `approval_ids`: expected approval IDs. A number smaller than
+    ///    2^53, and therefore representable as JSON. See Approval Management
+    ///    standard for full explanation. Must have same length as token_ids
+    /// * `memo` (optional): for use cases that may benefit from indexing or
+    ///    providing information for a transfer.
+    /// * `msg`: specifies information needed by the receiving contract in
+    ///    order to properly handle the transfer. Can indicate both a function to
+    ///    call and the parameters to pass to that function.
+
+    fn mt_batch_transfer_call(
+        &mut self,
+        receiver_id: AccountId,
+        token_ids: Vec<TokenId>,
+        amounts: Vec<U128>,
+        memo: Option<String>,
+        msg: String,
+    ) -> PromiseOrValue<Vec<U128>>;
+
+    /// Get the balance of an an account given token_id. For fungible token returns back amount, for
+    /// non fungible token it returns back constant 1.
```

Comment:
This seems weird -- I assume that this call must return 0 if someone doesn't own a particular fungible token, but this text makes it sound like it is an absolute constant, which I might expect for total supply, but not for balance of?

Reply:
We can broach this in a meeting, it's unclear if anyone feels like this doesn't make sense. In theory the comment is a bit of an over reach. I don't understand the intent of the  underlying implementation the multi token standard is a bit more free form. So there are fewer rules about what should and shouldn't happen. In this case  we're in alignment with erc-1155 suggestion. Although it's not a hard rule so I should probably just make it a suggestion. We're a little different because we use string token ids instead of numeric, but the same 
concept applies I think.
From ERC-1155: 
```
//
uint256 baseTokenNFT = 12345 << 128;
uint128 indexNFT = 50;

uint256 baseTokenFT = 54321 << 128;

balanceOf(baseTokenNFT, msg.sender); // Get balance of the base token for non-fungible set 12345 (this MAY be used to get balance of the user for all of this token set if the implementation wishes as a convenience).
balanceOf(baseTokenNFT + indexNFT, msg.sender); // Get balance of the token at index 50 for non-fungible set 12345 (should be 1 if user owns the individual non-fungible token or 0 if they do not).
balanceOf(baseTokenFT, msg.sender); // Get balance of the fungible base token 54321.
```

My vote is to just make the language more suggestive vs. an absolute rule.